### DISCUSSION TITLE
Migrating v0.11 state into the Reference Architecture

### DISCUSSION THREAD

**Discussion by iantybo** [gruntwork-io/knowledge-base#470] [Help] [answered]
Does Gruntwork provide any guidance for getting existing root modules and their states up to v1.2 if they began at v0.11? We would like these modules to eventually live in the Reference Architecture codebase. Is there any additional tooling outside of the `terraform` CLI helpers?

---

<ins datetime="2022-06-15T18:38:00Z">
  <p><a href="https://support.gruntwork.io/hc/requests/108784">Tracked in ticket #108784</a></p>
</ins>

**Comment by yorinasub17** [comment, marked as answer]
You can work through our [version specific upgrade guides](https://docs.gruntwork.io/guides/stay-up-to-date/) to work your way through the terraform versions, one minor version at a time.

We also have a guide on incrementally upgrading your terraform versions, assuming you are using `terragrunt`: see https://github.com/gruntwork-io/knowledge-base/discussions/95
